apps/fz_vpn/lib/fz_vpn/interface/wg_adapter/sandbox.ex

```diff
@@ -35,11 +39,17 @@ defmodule FzVpn.Interface.WGAdapter.Sandbox do
     {:reply, {:ok, Map.get(devices, name)}, devices}
   end
 
+  @impl GenServer
+  def handle_call({:list_devices}, _from, devices) do
+    {:reply, {:ok, Map.keys(devices)}, devices}
+  end
+
   @impl GenServer
   def handle_call({:set_device, config, name}, _from, devices) do
     public_key =
       if config.private_key do
-        Wireguardex.get_public_key(config.private_key)
+        {:ok, public_key} = Wireguardex.get_public_key(config.private_key)
```

Comment:
Remove the conditional and crash if no private key?

```suggestion
{:ok, public_key} = Wireguardex.get_public_key(config.private_key)        
```

Reply:
The private key can be optionally set in Wireguardex, so I was just mirroring that here.